packages/tools/release-utils-internal/src/flow.ts

```diff
@@ -0,0 +1,74 @@
+const steplogFnKey = Symbol("key for logger fn");
+const stepCountKey = Symbol("key for counter");
+
+const state = {};
+
+function createLogFn(): (...args: any[]) => void {
+    return (...args: any[]): void => {
+        console.info(`> Step ${state[stepCountKey]}:`, ...args);
+    };
+}
+
+function resetLogCount(): void {
+    state[stepCountKey] = 0;
+}
+
+function resetLogFn(): void {
+    state[steplogFnKey] = createLogFn();
+}
+
+function incStep(): void {
+    state[stepCountKey] += 1;
+}
+
+function clear(): void {
+    delete state[steplogFnKey];
+    delete state[stepCountKey];
+}
+
+function setup(): void {
+    resetLogFn();
+    resetLogCount();
+}
+
+function teardown(): void {
+    clear();
+}
+
+function log(...args: any[]): void {
+    const currentLogFn: (...args: any[]) => void = state[steplogFnKey];
+
+    if (typeof currentLogFn === "function") {
+        currentLogFn(...args);
+    } else {
+        console.warn("Can't find steplog fn in global scope. Did you forgot to run `start()`?");
+        console.info(...args);
+    }
+}
+
+// This function will create `locked` var in closure to
+function createRunStep(): (stepName: string, step: () => Promise<void>) => Promise<void> {
```

Comment:
If we go with that concept, we could pass a simple logger with (with console.{debug,info,log,warn,error}-like interface) to the `step` function and it could use it for logging. That way we don't spread around that `isDebug` check everywhere.